### PR TITLE
config command

### PR DESCRIPTION
**Pull request by fprimex**
Implement config file support and a command called `config` to manage it. Implements `curl` config file generation, which can contain the token so as to not expose the token value in the process list. If the `curl` config file does not exist, but the `.terraformrc` does exist, then an attempt is made to generate the `curl` config from the token value in the `.terraformrc` config.

```
$ rm ~/.tfe-cli*
rm: /Users/brent/.tfe-cli*: No such file or directory

$ tfe config
TFE_ORG=""
TFE_WORKSPACE=""
TFE_URL="https://app.terraform.io"
# TFE_TOKEN set by /Users/brent/.tfe-cli-curl generated from /Users/brent/.terraformrc

$ tfe config -name a/b
TFE_ORG="a"
TFE_WORKSPACE="b"
TFE_URL="https://app.terraform.io"
# TFE_TOKEN set by /Users/brent/.tfe-cli-curl

$ cat ~/.tfe-cli-curl 
--header "Authorization: Bearer REDACTED"

$ tfe config -name a/b -write
TFE_ORG="a"
TFE_WORKSPACE="b"

$ cat ~/.tfe-cli
TFE_ORG="a"
TFE_WORKSPACE="b"

$ echo "SOME_OTHER_VAR=VALUE" >> ~/.tfe-cli

$ tfe config -name c/d -write
TFE_ORG="c"
TFE_WORKSPACE="d"
SOME_OTHER_VAR=VALUE

$ cat ~/.tfe-cli
TFE_ORG="c"
TFE_WORKSPACE="d"
SOME_OTHER_VAR=VALUE

$ tfe config -token ANOTHERTOKEN -write
TFE_ORG="c"
TFE_WORKSPACE="d"
SOME_OTHER_VAR=VALUE
# Wrote /Users/brent/.tfe-cli-curl

$ cat ~/.tfe-cli-curl 
--header "Authorization: Bearer ANOTHERTOKEN"
```

```
$ rm ~/.tfe-cli*

$ tfe config -tfe-org brentwoodruff -write
TFE_ORG="brentwoodruff"

$ ls ~/.tfe-cli*
/Users/brent/.tfe-cli      /Users/brent/.tfe-cli-curl

$ tfe workspace list
  tfe_demo
  uses-custom-tf-provider-1
  uses-custom-tf-provider-2
  uses-my-tfvars
  uses_global_tfvars
  ws-baz

$ tfe config -tfe-workspace ws-baz
TFE_ORG="brentwoodruff"
TFE_WORKSPACE="ws-baz"
TFE_URL="https://app.terraform.io"
# TFE_TOKEN set by /Users/brent/.tfe-cli-curl

$ tfe config -tfe-workspace ws-baz -write
TFE_ORG="brentwoodruff"
TFE_WORKSPACE="ws-baz"

$ tfe pullvars
foo = "bar"

$ tfe workspace delete
Deleted brentwoodruff/ws-baz
```

### CODE REVIEW

File: bin/tfe [hunk 1]
```diff
@@ -75,6 +75,30 @@ cleanup () {
...
+    if ! echo > "$1"; then
+        echoerr "Error: cannot write to curl config file:"
+        echoerr "$1"
+        return 1
+    fi
+
+    if ! chmod 600 "$1"; then
+        echoerr "WARNING: unable to set permissions on curl config file:"
+        echoerr "chmod 600 $1"
+    fi
```
Comment (brianshumate) [inline bin/tfe]: Does this need a `return 1` also?

Comment (fprimex) [reply bin/tfe]: I originally had it `exit 1`, but then I decided that maybe a warning was better. The permissions could be fixed by the user after the command completed.

Comment (fprimex) [reply bin/tfe]: (so it doesn't exit or return 1, just prints to stderr)

File: bin/tfe [hunk 2]
```diff
@@ -388,21 +407,61 @@ else
...
+if [ -f "$tfe_config" ]; then
+    echodebug "[DEBUG] Loading $tfe_config"
+    . "$tfe_config"
+else
+    echodebug "[DEBUG] No tfe config file found"
+fi
+
 # Allow setting the URL to TFE with TFE_URL
-tfe_address=${TFE_URL:-"https://app.terraform.io"}
+tfe_default_address="https://app.terraform.io"
```
Comment (mwhooker) [inline bin/tfe]: I like to use `readonly` for defaults, but don't think it's necessary

`readonly tfe_default_address="https://app.terraform.io"`

Comment (fprimex) [reply bin/tfe]: Thanks for the suggestion. I gave the defaults the readonly treatment. There might be other good places to use that, such as locking the "global" vars like `tfe_org`. I'll look at that later.

**Comment by fprimex** [comment]
Thanks for the feedback. Merging to `devel`, so there'll be more time for testing and whatnot before going to master.

**Pull request merged**